spring-context/src/main/java/org/springframework/context/ApplicationEventPublisherAware.java

```diff
@@ -37,6 +38,6 @@
 	 * Invoked before ApplicationContextAware's setApplicationContext.
 	 * @param applicationEventPublisher event publisher to be used by this object
 	 */
-	void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher);
+	void setApplicationEventPublisher(@Nullable ApplicationEventPublisher applicationEventPublisher);
```

Comment:
Let's leave the `Aware` interfaces as non-null since the container will never call them with a non-null reference. For this particular case, we could keep the `AbstractBrokerMessageHandler.setApplicationEventPublisher` implementation as nullable but leave the interface untouched.